Xamarin.Forms.Core/IndicatorsView.cs

```diff
@@ -0,0 +1,285 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	public enum IndicatorShape
+	{
+		Circle,
+		Square
+	}
+
+	public class IndicatorsView : StackLayout
+	{
+		public static readonly BindableProperty PositionProperty = BindableProperty.Create(nameof(Position), typeof(int), typeof(IndicatorsView), default(int), BindingMode.TwoWay, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
```

Comment:
use direct cast

Reply:
use a lambda, or a static method, but avoid `{}` in BP definition

Reply:
> use direct cast

is it code style or there are any reasons?

In terms of "code readability", i think the best idea would be adding an extension method (As<T>(object o) where T: class)

and using it: bindable.As<IndicatorsView>()?.ResetIndicatorsStyles()

anyway I will amend

Reply:
should be:

```suggestion
			((IndicatorsView)bindable).ResetIndicatorsStyles();
```

like it's done everywhere else. I prefer a failing cast exception than continuing with a property being set on the wrong control

Reply:
makes sense

Reply:
done